### PR TITLE
Upgrade Prometheus packages and related dependencies

### PR DESCRIPTION
**Pull request by santihernandezc**
**What this PR does / why we need it**:
It upgrades Prometheus' `alertmanager` and `prometheus` packages to use new functionality like using their `TemplateExpander` and being able to return an error when a key is missing. It also updates related dependencies.

**Which issue(s) this PR fixes**:
Not directly, but it's related to issue [#36401](https://github.com/grafana/grafana/issues/36401)

### CODE REVIEW

**Comment by gotjosh** [comment]
I have verified that:

- `google.golang.org/api v0.56.0`
- `github.com/cespare/xxhash/v2 v2.1.2`
- `github.com/aws/aws-sdk-go v1.40.37`
- `github.com/go-openapi/strfmt v0.20.2`
- `golang.org/x/net v0.0.0-20210903162142-ad29c8ab022f`
- `golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f`
- `google.golang.org/api v0.56.0`

All come from Prometheus (you can refer to https://github.com/prometheus/prometheus/pull/9308 for more) but there are a few others that I don't know where they come from e.g. `	github.com/denisenkom/go-mssqldb` or `github.com/influxdata/influxdb-client-go/v2` - can you verify if these were upgraded by mistake?

**Comment by santihernandezc** [comment]
> I have verified that:
> 
> * `google.golang.org/api v0.56.0`
> * `github.com/cespare/xxhash/v2 v2.1.2`
> * `github.com/aws/aws-sdk-go v1.40.37`
> * `github.com/go-openapi/strfmt v0.20.2`
> * `golang.org/x/net v0.0.0-20210903162142-ad29c8ab022f`
> * `golang.org/x/oauth2 v0.0.0-20210819190943-2bc19b11175f`
> * `google.golang.org/api v0.56.0`
> 
> All come from Prometheus (you can refer to [prometheus/prometheus#9308](https://github.com/prometheus/prometheus/pull/9308) for more) but there are a few others that I don't know where they come from e.g. ` github.com/denisenkom/go-mssqldb` or `github.com/influxdata/influxdb-client-go/v2` - can you verify if these were upgraded by mistake?

The following packages are updated after upgrading Prometheus via the command `go get github.com/prometheus/prometheus@bd217c5`:
- `github.com/denisenkom/go-mssqldb` 
- `github.com/go-openapi/strfmt`
- `github.com/influxdata/influxdb-client-go/v2`
- `github.com/matttproud/golang_protobuf_extensions`
- `golang.org/x/net`
- `golang.org/x/oauth2`
- `google.golang.org/api`

**Review by gotjosh — APPROVED** [review]
You'll need to rebase but overall this LGTM. 

I noticed that influx-db and MySQL are indirect dependencies.

File: CHANGELOG.md [hunk 1]
```diff
@@ -58,20 +57,17 @@
...
+Issue [#38078](https://github.com/grafana/grafana/issues/38078)
 
-Panel queries and/or annotation queries that used more than one statistic will be converted into one query/annotation per statistic. In case an alerting rule was based on a query row that had more than one statistic, it would now be based only on the first statistic for that query row. New alerting rules will not be created for migrated queries. Please note that in most cases it would not make sense to have an alerting rule that is based on multiple statistics anyway.  Issue [#36925](https://github.com/grafana/grafana/issues/36925)
+Panel queries and/or annotation queries that used more than one statistic will be converted into one query/annotation per statistic. In case an alerting rule was based on a query row that had more than one statistic, it would now be based only on the first statistic for that query row. New alerting rules will not be created for migrated queries. Please note that in most cases it would not make sense to have an alerting rule that is based on multiple statistics anyway. Issue [#36925](https://github.com/grafana/grafana/issues/36925)
```
Comment (achatterjee-grafana) [inline CHANGELOG.md:67]:
```suggestion
Panel and annotation queries using more than one statistic now converts into one query/annotation per statistic. Alerting rules based on a query row with multiple statistics now considers only the first statistic from that query row. New alerting rules are not created for migrated queries. Usually, it does not make sense to have an alerting rule based on multiple statistics. Issue [#36925](https://github.com/grafana/grafana/issues/36925)
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

**Review by achatterjee-grafana — APPROVED** [review]

**Pull request merged**